src/dal/dal.py

```diff
@@ -10,9 +8,9 @@
 
 class LibraryItemRepository(ABC):
     @abstractmethod
-    def update_item_status(self, isbn: str, is_borrowed: bool) -> None:
+    async def update_item_status(self, isbn: str, is_borrowed: bool) -> None:
         """
-        Update the status of a library item by its ISBN.
+        Asynchronously update the status of a library item by its ISBN.
```

Comment:
Stop adding asynchronously everywhere.
Obviously it is async since you used `async def`